### PR TITLE
initial setup with some questions

### PR DESCRIPTION
**Pull request by leuthera**
hope this helps you get started

### CODE REVIEW

**Comment by monkey3310** [comment]
Hey @leuthera, thanks for your contribution!
It happens that I've already created `javascript` file on another branch (its already merged to `master` - and some of the questions are already there (without answers though!)

Can you migrate your questions to the new file (and follow notation I proposed there? Thanks! 👍

**Comment by leuthera** [comment]
@monkey3310 ok, i hope to get it done today.

**Comment by leuthera** [comment]
@monkey3310 i've migrated the question to your desired design.

File: tech/javascript.MD [hunk 1]
```diff
@@ -22,103 +22,196 @@ In the **global execution context (outside of any function)**, `this` refers to
...
+</details>
+
+### 15. What is Hoisting?
+:star: _Junior_
+<details>
+    <summary>
+        Answer
+    </summary>
+
+Means that you can use functions and "var"-s before you declare them
```
Comment (monkey3310) [inline tech/javascript.MD]: This is actually outcome of JS Hoisting: 
> Hoisting is JavaScript's default behavior of moving all declarations to the top of the current scope (to the top of the current script or the current function).

Also, mention the `let` and `const` keywords and and this:
> JavaScript only hoists declarations, not initializations.

File: tech/javascript.MD [hunk 2]
```diff
@@ -22,103 +22,196 @@ In the **global execution context (outside of any function)**, `this` refers to
...
+<details>
+    <summary>
         Answer
     </summary>
+
+Switches to strict mode which helps to prevent common errors like using unsafe operators
+
+</details>
+
+### 17. When to use .bind()?
```
Comment (monkey3310) [inline tech/javascript.MD]: Could you change the question to _What is `bind()` and when we use it?_ + provide more extended explanation

File: tech/javascript.MD [hunk 3]
```diff
@@ -22,103 +22,196 @@ In the **global execution context (outside of any function)**, `this` refers to
...
+</details>
+
+### 18. Explain higher order function?
+:star: _Senior_
+<details>
+    <summary>
+        Answer
+    </summary>
+
+Function that will take a function as rgument or return a new function
```
Comment (monkey3310) [inline tech/javascript.MD]: typo here 😃

File: tech/javascript.MD [hunk 4]
```diff
@@ -22,103 +22,196 @@ In the **global execution context (outside of any function)**, `this` refers to
...
+### 19. Explain map, filter, reduce and when to use it?
+:star: _Senior_
+<details>
+    <summary>
+        Answer
+    </summary>
+
+map - to iterate over an array and return a new one
+filter - to filter an array and return a new filtered one
+reduce - to do what ever you want with an array and return whatever you want, e.g. filter some values and summarize them
```
Comment (monkey3310) [inline tech/javascript.MD]: for the `reduce` I would say that it takes a reducer function that will be evaluated against each element of an array and returns output value

File: tech/javascript.MD [hunk 5]
```diff
@@ -22,103 +22,196 @@ In the **global execution context (outside of any function)**, `this` refers to
...
 
 ### 2. What is the difference between `let` and `var` and `const`?
 :star: _Junior_
 <details>
-    <summary>            
+    <summary>
         Answer
     </summary>
+
+The most difference is scoping.
```
[the remark this answers is not in the document]
Comment (leuthera) [reply tech/javascript.MD]: Maybe its about both, assignment and scoping:
```
for ( var i=0; i<2; i++ ) {} console.log(i) // exists outside the blockscope
for ( let i=0; i<2; i++ ) {} console.log(i) // only exists inside the blockscope
for ( const i=0; i<2; i++ ) {} console.log(i) // error reassignment, but only on top-level
for ( const cnt={i:0}; cnt.i<2; cnt.i++ ) { console.log(cnt.i) } // only exists inside the blockscope
```
I like your explanation and would just add that const needs initialisiation and var gets hoisted and the 2 different scopes.

**Comment by leuthera** [comment]
I'am done for today, if you have more things todo pls just write or doit youself :P. I'll be back tomorrow.

**Pull request merged**